pandas/core/reshape/merge.py

```diff
@@ -492,6 +492,10 @@ def __init__(self, left, right, how='inner', on=None,
                  left_index=False, right_index=False, sort=True,
                  suffixes=('_x', '_y'), copy=True, indicator=False,
                  validate=None):
+        if isinstance(left, Series):
```

Comment:
if u are actually going to do this
then you need to strip a ton of logic from below

Reply:
Can you clarify this comment? What logic would need to be stripped?

Currently, this just raises when you pass a Series (see line 525 a bit below)

Reply:
Thanks both - will make edits to this based on @jreback 's clarification

Reply:
nvm on the code removal, we do extension promotion in ``concat``, but nothing here.